### ISSUE TITLE
Add temperature, top-k and top-p logits processors

### ISSUE THREAD

**Issue by SimJeg** [outlines-dev/outlines#335]
Hello,

In [outlines.text.generate.sample](https://github.com/outlines-dev/outlines/blob/main/outlines/text/generate/sample.py), only `greedy` and `multinomial` methods are available. Would it be possible to implement some classical features in the `multinomial` function such as `top_k`, `top_p` or `temperature` ? 

Thanks for your work,
Simon

**Comment by brandonwillard** [comment]
Yes, definitely!

**Comment by rlouf** [comment]
> Also, we need to add beam search.

Tracked by #258

**Comment by rlouf** [comment]
#533 introduces a small refactor of the samplers that would allow to pass the `top_k` and `top_p` arguments to the `multinomial` sampler. We could pass the temperature using the same mechanism as well

**Comment by rlouf** [comment]
top-k was added in #642, and I opened separate issues for the other requests: #645 and #643.
